### ISSUE TITLE
Strange behaviour `no-missing-eof-newline`

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#562]
```no-missing-eof-newline: 2```
I have file _variables.scss:
```scss

```
With one empty line. Bug i have error
```bash
1:1	⚠  Unexpected missing newline at end of file (no-missing-eof-newline) [stylelint]
```

**Comment by davidtheclark** [comment]
You do not have an empty line at the end of the file, right? Your file is just an empty string? In this case, *should* the rule ignore the file?

**Comment by evilebottnawi** [comment]
@davidtheclark i have file with one empty line. Because i have editorconfig with config `insert_final_newline = true` and all files have new line. Many developers recommend use this feature and insert final newline in file. But file empty and my IDE with editorconfig add newline to file. I think rule should ignore this situation.

**Comment by davidtheclark** [comment]
Just to clarify: I don't think you have a newline in your file. Can you get
a JSON string version of the file and verify that it has a newline?

The reason is that we do have a test testing for a file containing nothing
but a newline, and it passes.

If I'm right, what you're requesting is that the rule accept files that are
just empty strings.

**Comment by evilebottnawi** [comment]
@davidtheclark you are right my file not contain newline. It's just an empty file. Maybe it would be right to ignore the empty files for this rule.

**Comment by davidtheclark** [comment]
If anybody has a chance, it would be good to search out precedent --- like,
does ESLint (or other JS linters with the same kind of rule) allow empty
files?

**Comment by evilebottnawi** [comment]
@davidtheclark i use eslint, he not throw error on empty files.

**Comment by evilebottnawi** [comment]
@davidtheclark `eol-last: 2` 
http://eslint.org/docs/rules/eol-last
`Trailing newlines in non-empty files are a common UNIX idiom.`

**Comment by davidtheclark** [comment]
Sounds good. Created https://github.com/stylelint/stylelint/issues/564.
